nemo/core/classes/common.py

```diff
@@ -44,12 +44,18 @@ def output_types(self) -> Optional[Dict[str, NeuralType]]:
     def _validate_input_types(self, **kwargs):
         # TODO: Properly implement this
         if self.input_types is not None:
-            if len(kwargs) != len(self.input_types):
-                raise TypeError(
-                    "Number of input arguments provided ({}) is not as expected ({})".format(
-                        len(kwargs), len(self.input_types)
+            total_input_types = len(self.input_types)
+            mandatory_input_types = len(
+                [type_val for type_key, type_val in self.input_types.items() if not type_val.optional]
+            )
+
+            if len(kwargs) != total_input_types:
+                if len(kwargs) != mandatory_input_types:
+                    raise TypeError(
+                        "Number of input arguments provided ({}) is not as expected ({})".format(
+                            len(kwargs), len(self.input_types)
+                        )
```

Comment:
This if can be condensed

Reply:
Good point. Done (in next commit)

Reply:
Spoke too soon. Cant condense it cause we need to disparate checks - 1 in case total_input_types match, and other if mandatory types match. If we use `and` or `or` we will always fail one of the two tests.